crates/backend/src/ast.rs

```diff
@@ -110,91 +123,144 @@ impl Hash for ImportModule {
     }
 }
 
+/// The type of item being imported
 #[cfg_attr(feature = "extra-traits", derive(Debug))]
 #[derive(Clone)]
 pub enum ImportKind {
+    /// Importing a function
     Function(ImportFunction),
+    /// Importing a static value
     Static(ImportStatic),
+    /// Importing a type/class
     Type(ImportType),
+    /// Importing a JS enum
     Enum(ImportEnum),
 }
 
+/// A function being imported from JS
 #[cfg_attr(feature = "extra-traits", derive(Debug))]
 #[derive(Clone)]
 pub struct ImportFunction {
+    /// The full signature of the function
     pub function: Function,
+    /// The name rust code will use
     pub rust_name: Ident,
+    /// The type being returned
     pub js_ret: Option<syn::Type>,
+    /// Whether to catch JS exceptions
     pub catch: bool,
+    /// Whether the function is variadic on the JS side
     pub variadic: bool,
+    /// Whether the function should use structural type checking
     pub structural: bool,
+    /// For testing purposes
     pub assert_no_shim: bool,
+    /// The kind of function being imported
     pub kind: ImportFunctionKind,
+    /// The shim name to use in the generated code
```

Comment:
FWIW the "shim" here I think could use more explanation. It's not 100% clear from nothing what the shim even is here I think.